### DISCUSSION TITLE
command line interface for configuration of SOZI

### DISCUSSION THREAD

**Discussion by nnako** [senshu/Sozi#639] [Ideas]
Hi there,

I have a use case for SOZI where I frequently add "interactivity" to a big SVG map. As the contents of the SVG map do only change in minor parts (e.g. text contents within a frame but not any frame location in the map), I was wondering if I could skip the editing using the SOZI EDITOR APPLICATION. I'd rather use a kind of predefined configuration which I apply by using e.g. a command line interface to SOZI.

Is such a command line interface already provided? Or maybe, another kind of automation interface to tell SOZI to apply the same kind of configuration whenever re-creating a presentation?

Here, you see a concept image where I show the different steps from the original SVG file read-in (on the left) until the resulting presentation HTML file (on the right). Inbetween, I use Inkscape to pre-configure the SVG image and SOZI to create the frames and zoom levels:

![image](https://user-images.githubusercontent.com/5845602/179361904-1a65a912-e381-45a4-9e5c-d081080b07d0.png)

The "CREATOR" on the bottom of the concept image is not using the editor but rather a command line interface (CLI) to tell SOZI how to prepare the presentation HTML file. This would be my request.

Any ideas how use SOZI in a sort of automated / preconfigured way?

**Comment by aumouvantsillage** [comment]
Thanks for suggesting this use case. Currently, Sozi is not usable from the command line but I agree that there are common operations that would not need the GUI. See #585 for another example.

> As the contents of the SVG map do only change in minor parts (e.g. text contents within a frame but not any frame location in the map)

In this specific situation, a workaround would be to make a copy of the original JSON file, with the name of the new SVG.

1. Create `first-presentation.svg` with Inkscape.
2. Open `first-presentation.svg` with Sozi and create some frames. The presentation data are saved in `first-presentation.sozi.json`.
3. Modify `first-presentation.svg` with Inkscape and save it as `second-presentation.svg`.
4. Duplicate `first-presentation.sozi.json` as `second-presentation.sozi.json`.
5. Open `second-presentation.svg` in Sozi.

**Comment by Ageman** [comment]
I have a similar request: 
I am changing my svg's frequently and want to rebuild all presentations nightly in a Jenkins Pipeline. It would be great to have Command Line Parameters to just call sozi with the svg and it just build the html without opening the Editor (in fact I do not even have X in my Jenkins environment)
Something like
`sozi --build my.svg`

**Comment by hkiel** [comment]
I would be interested in this feature as well. Shouldn't bee too hard to implement!?
